### PR TITLE
Cc user rename

### PR DESCRIPTION
**Pull request by carlymclift**
### What’s this PR do?
* allows the method getFriendsName() in User class to work independently of userRepo
* adds sad paths to the user and UserRepo test suites

### Where should the reviewer start?
user.js, then testing files

### How should this be manually tested?
`npm test`

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -69,11 +69,11 @@ function getUsersRecentDate(id, dataSet) {
...
-function makeRandomDate(userStorage, id, dataSet) {
-  // gives a random date (used where?)
-  var sortedArray = userStorage.makeSortedUserArray(id, dataSet); // sorts data set by date from recent
-  return sortedArray[Math.floor(Math.random() * sortedArray.length + 1)].date;
-}
+// function makeRandomDate(userStorage, id, dataSet) {
+//   // gives a random date (used where?)
+//   var sortedArray = userStorage.makeSortedUserArray(id, dataSet); // sorts data set by date from recent
+//   return sortedArray[Math.floor(Math.random() * sortedArray.length + 1)].date;
+// }
```
Comment (jordy1611) [inline src/scripts.js:76]: Why did we comment this out

Comment (relyt4me) [reply src/scripts.js:76]: We are removing the random week from the page as it is not used in a user friendly way

File: src/User.js [hunk 2]
```diff
@@ -11,18 +9,17 @@ class User {
...
-  // getNamesOfFriends()
-  getFriendsNames(userStorage) {
-    return this.friends.map((friendId) => (userStorage.getDataFromID(friendId).name));
+  getFriendsNames(users) {
+    let friendNames = this.friends.map(friendID => {
+      return users.find(user => {
+        return user.id === friendID
+      }).name;
+    })
+    return friendNames
```
Comment (relyt4me) [inline src/User.js:22]: I like this way of logic and naming of users instead of userStorage.

File: src/User-repo.js [hunk 3]
```diff
@@ -29,11 +20,15 @@ class UserRepo {
   }
 
   /* 
-  Methods below shouldn't really be in the UserRepo, 
-  they are better off in scripts or other files
-  however they used everywhere and aren't easy to move
+  Methods that shouldn't be in userRepo below:
```
Comment (relyt4me) [inline src/User-repo.js:23]: I cannot wait for these to all be gone 👍

**Pull request merged**